test/unit/changePasswordWebviewController.test.ts

```diff
@@ -0,0 +1,173 @@
+
+import * as vscode from "vscode";
+import * as sinon from "sinon";
+import sinonChai from "sinon-chai";
+import * as chai from "chai";
+
+import { expect } from "chai";
+
+import * as jsonRpc from "vscode-jsonrpc/node";
+
+import { ChangePasswordWebviewController } from "../../src/controllers/changePasswordWebviewController";
+import { ChangePasswordService } from "../../src/services/changePasswordService";
+import {
+    ChangePasswordWebviewRequest,
+    CancelChangePasswordWebviewNotification,
+} from "../../src/sharedInterfaces/changePassword";
+import * as LocConstants from "../../src/constants/locConstants";
+import {
+    stubTelemetry,
+    stubVscodeWrapper,
+    stubWebviewConnectionRpc,
+    stubWebviewPanel,
+} from "./utils";
+import VscodeWrapper from "../../src/controllers/vscodeWrapper";
+import { IConnectionInfo } from "vscode-mssql";
+import { Logger } from "../../src/models/logger";
+import * as utils from "../../src/utils/utils";
+
+chai.use(sinonChai);
+
+suite("ChangePasswordWebviewController", () => {
+    let sandbox: sinon.SinonSandbox;
+    let mockContext: vscode.ExtensionContext;
+    let vscodeWrapperStub: sinon.SinonStubbedInstance<VscodeWrapper>;
+    let changePasswordServiceStub: sinon.SinonStubbedInstance<ChangePasswordService>;
+    let requestHandlers: Map<string, (password: string) => Promise<unknown>>;
+    let notificationHandlers: Map<string, () => void>;
+    let connectionStub: jsonRpc.MessageConnection;
+    let createWebviewPanelStub: sinon.SinonStub;
+    let panelStub: vscode.WebviewPanel;
+    let controller: ChangePasswordWebviewController;
+    let credentials: IConnectionInfo;
+
+    setup(() => {
+        sandbox = sinon.createSandbox();
+        stubTelemetry(sandbox);
+
+        const loggerStub = sandbox.createStubInstance(Logger);
+        sandbox.stub(Logger, "create").returns(loggerStub);
+
+        sandbox.stub(utils, "getNonce").returns("test-nonce");
+        sandbox.stub(LocConstants.Connection, "ChangePassword").value("Change Password");
+
+        const connection = stubWebviewConnectionRpc(sandbox);
+
+        requestHandlers = connection.requestHandlers;
+        notificationHandlers = connection.notificationHandlers;
+
+        connectionStub = connection.connection;
+        sandbox
+            .stub(jsonRpc, "createMessageConnection")
+            .returns(connectionStub as unknown as jsonRpc.MessageConnection);
+
+        panelStub = stubWebviewPanel(sandbox);
+
+        createWebviewPanelStub = sandbox
+            .stub(vscode.window, "createWebviewPanel")
+            .callsFake(() => panelStub);
+
+        sandbox.stub(vscode.window, "onDidChangeActiveColorTheme").returns({
+            dispose: sandbox.stub(),
+        } as vscode.Disposable);
+        sandbox.stub(vscode.window, "activeColorTheme").value({
+            kind: vscode.ColorThemeKind.Light,
+        } as vscode.ColorTheme);
+        sandbox.stub(vscode.window, "showInformationMessage").resolves(undefined);
```

Comment:
are these actually necessary?  We should be sure to clean up extra crap that AI throws in there as much as possible.